src/libaktualizr/campaign/campaign.h

```diff
@@ -30,18 +30,30 @@ static inline Cmd cmdFromName(const std::string &name) {
 // Out of uptane concept: update campaign for a device
 class Campaign {
  public:
+  Campaign() = default;
+  Campaign(std::string idIn, std::string nameIn, int64_t sizeIn, bool autoAcceptIn, std::string descriptionIn,
```

Comment:
What is this constructor being used for? If it's really needed, it seems like it might be easier to provide a constructor with json input instead.

Reply:
Yep, also not clear for me. And it looks like the static fromJson method can rather be a ctor.

Reply:
I usually don't like to put deserialization methods in pure constructors as `fromJson` seems clearer to me. Once another mean of creating an object is needed, there would be no risk of confusion with a simple constructor.

But that's a minor style preference, the constructor solution seems fine here.